app/modules/intelligence/agents/agents/rag_agent.py

```diff
@@ -74,24 +79,22 @@ def __init__(self, sql_db, llm, mini_llm, user_id):
         self.llm = llm
         self.mini_llm = mini_llm
         self.user_id = user_id
+        self.prompt_service = PromptService(self.sql_db)
 
     async def create_agents(self):
+        llm_provider_service = LLMProviderService.create(self.sql_db, self.user_id)
+        preferred_llm, _ = await llm_provider_service.get_preferred_llm(self.user_id)
+        agent_prompt = await self.prompt_service.get_prompts(
+            "debug_rag_query_agent",
```

Comment:
is this supposed to be the debug agent prompt?  Please crosscheck all the agents again